spec/integration/support/serve_examples.rb

```diff
@@ -47,13 +47,17 @@ def test_basic_html(port)
       @machinery.run_command("pkill -f 'machinery serve' --signal 9")
     end
 
-    it "makes the system description HTML and extracted files available at the specified port" do
+    before(:each) do
       @machinery.inject_directory(
         system_description_dir,
         machinery_config[:machinery_dir],
         owner: machinery_config[:owner],
         group: machinery_config[:group]
       )
+    end
+
+    it "makes the system description HTML and extracted files available at the specified port and " \
+      "checks if the port is already in use when we run that command twice" do
```

Comment:
That was actually just a mistake. We don't need the `and checks if the port is already in use when we run that command twice` part anymore since I implemented an extra test for that. Removed.